bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/GenericComplexType.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.smarthome.core.library.types;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Formatter;
+import java.util.List;
+
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * This type can be used for items that are dealing with telephony functionality.
+ *
+ * @author Thomas.Eichstaedt-Engelen / Initially CallType
+ * @author Gaël L'hopital - port to Eclipse SmartHome
+ *
+ */
+public class GenericComplexType implements Command, State {
+
+    List<String> typeDetails;
+
+    public GenericComplexType(StringType... rows) {
+        typeDetails = new ArrayList<String>();
+        for (StringType row : rows) {
+            typeDetails.add(row.toString());
+        }
+    }
+
+    public GenericComplexType(String... rows) {
+        typeDetails = Arrays.asList(rows);
+    }
+
+    /**
+     * Deserialize the input string,
+     * splitting it on every semicolon not preceeded by a backslash
+     */
+    public GenericComplexType(String serialized) {
+        String[] rows = serialized.split("(?<!\\\\);");
+        typeDetails = new ArrayList<String>();
+        for (String row : rows) {
+            typeDetails.add(row.replace("\\;", ";"));
+        }
+    }
+
+    public String getValue(final int index) {
```

Comment:
Shouldn't we start the index by 0 as usual?
Any reason to start the index by 1 here?